pkg/controller/kabaneroplatform/featured_stacks.go

```diff
@@ -137,3 +148,53 @@ func featuredStacks(k *kabanerov1alpha2.Kabanero, cl client.Client) (map[string]
 
 	return stackMap, nil
 }
+
+// Cleans up current stacks based on desired state. Current stack versions with the active state must be preserved.
+func preProcessCurrentStacks(ctx context.Context, k *kabanerov1alpha2.Kabanero, cl client.Client, indexStackMap map[string][]kabanerov1alpha2.StackVersion) error {
+	deployedStacks := &kabanerov1alpha2.StackList{}
+	err := cl.List(ctx, deployedStacks, client.InNamespace(k.GetNamespace()))
+	if err != nil {
+		return err
+	}
+
+	// Compare the list of currently deployed stacks and the stacks in the index.
+	for _, deployedStack := range deployedStacks.Items {
+		iStackList, _ := indexStackMap[deployedStack.GetName()]
+		newStackVersions := []kabanerov1alpha2.StackVersion{}
+		for _, dStackVersion := range deployedStack.Spec.Versions {
+			deployedStackVersionMatchIndex := false
+			// Keep the stacks with matching versions. The caller will do the updates if necessary.
+			for _, iStack := range iStackList {
+				if dStackVersion.Version == iStack.Version {
+					deployedStackVersionMatchIndex = true
+					newStackVersions = append(newStackVersions, dStackVersion)
+					break
+				}
+			}
+
+			// Keep any stack versions that have a desired state of active.
+			if !deployedStackVersionMatchIndex && dStackVersion.DesiredState == kabanerov1alpha2.StackDesiredStateActive {
```

Comment:
Ok so now that I see this too, I wonder if I just did a poor job of explaining desiredState.  I think that if desiredState is set to anything (not just `active`) that the operator should not delete it.